lib/livyClient/lib/client.js

```diff
@@ -62,27 +99,43 @@ class LivyClient {
         return undefined;
     }
 
-    /** Returns interactive sessions
+    /** Return session or batch information by id
+     * @param {string} batchOrSession - either 'batch' or 'session'
+     * @param {number} id - id of session to get information on
+     * @param {funcftion} callback - callback
+     * @return {undefined}
+     */
+    getSessionOrBatch(batchOrSession, id, callback) {
+        const pathPrefix = inputCheck(batchOrSession, id, callback);
+        this._request('GET', `${pathPrefix}/${id}`);
+        return undefined;
+    }
+
+    /** Returns interactive sessions or batches
+     * @param {string} batchOrSession - either 'batch' or 'session'
      * @param {number} [startIndex] - index to start listing
-     * @param {number} [numOfSessions] - number of sessions to
+     * @param {number} [numOfSessionOrBatch] - number of sessions to
      * return
      * @param {function} callback - callback
      * @return {undefined}
      */
-    getSessions(startIndex, numOfSessions, callback) {
-        assert(typeof callback === 'function', 'callback must be a function');
+    getSessionsOrBatches(batchOrSession, startIndex, numOfSessionOrBatch,
+    callback) {
+        inputCheck(batchOrSession, null, callback);
```

Comment:
doesn't inputCheck return whether batch or session so don't have to check again below?

Reply:
Right now I'm returning the strings 'batch' or 'session', to use to check which request prefix I will be using ('batch' -> '/batches' vs 'session' -> '/sessions'). It makes more sense though to simply return the correct prefix to avoid the second check, thanks for pointing it out!